relay-server/src/services/processor/ourlog.rs

```diff
@@ -112,13 +131,54 @@ fn scrub(
     Ok(())
 }
 
+#[cfg(feature = "processing")]
+fn process_attribute_types(ourlog: &mut OurLog) {
+    let Some(attributes) = ourlog.attributes.value_mut() else {
+        return;
+    };
+
+    let attributes = attributes.iter_mut().map(|(_, attr)| attr);
+
+    for attribute in attributes {
+        use OurLogAttributeType::*;
+
+        let Some(inner) = attribute.value_mut() else {
+            continue;
+        };
+
+        match (&mut inner.value.ty, &mut inner.value.value) {
+            (Annotated(Some(Boolean), _), Annotated(Some(Value::Bool(_)), _)) => (),
+            (Annotated(Some(Integer), _), Annotated(Some(Value::I64(_)), _)) => (),
+            (Annotated(Some(Integer), _), Annotated(Some(Value::U64(_)), _)) => (),
+            (Annotated(Some(Double), _), Annotated(Some(Value::I64(_)), _)) => (),
+            (Annotated(Some(Double), _), Annotated(Some(Value::U64(_)), _)) => (),
+            (Annotated(Some(Double), _), Annotated(Some(Value::F64(_)), _)) => (),
+            (Annotated(Some(String), _), Annotated(Some(Value::String(_)), _)) => (),
+            (Annotated(ty_opt @ Some(Unknown(_)), ty_meta), _) => {
+                ty_meta.add_error(ErrorKind::InvalidData);
+                ty_meta.set_original_value(ty_opt.take());
+            }
+            (Annotated(Some(_), _), Annotated(value @ Some(_), value_meta)) => {
+                value_meta.add_error(ErrorKind::InvalidData);
+                value_meta.set_original_value(value.take());
+            }
+            (Annotated(None, _), _) | (_, Annotated(None, _)) => {
+                let original = attribute.value_mut().take();
+                attribute.meta_mut().add_error(ErrorKind::MissingAttribute);
+                attribute.meta_mut().set_original_value(original);
+            }
```

Comment:
@Dav1dde I was relying on overwriting to `Unknown` here so when it got transformed into LogKafkaMessage we can omit the attributes from the map entirely, not sure if there is a better way to drop Annotated with errors before the serialization? I'm pretty sure the existing consumer will throw errors if we emit an empty object. 

It's the only thing failing at the moment on the python integration test.

Reply:
We can do the same we do in case of `None` and set the value in the map to `null` instead, e.g. with:

```
                let original = attribute.value_mut().take();
                attribute.meta_mut().add_error(ErrorKind::MissingAttribute);
                attribute.meta_mut().set_original_value(original);
```

But this is turning into a bigger problem, we're working here on wire protocol how SDK and Relay to Relay communication is taking place, and on that protocol nulls need to be expected anywhere. E.g. the entire errors product from start to finish was built on this.

Reply:
Updated it so the entire attribute is removed.